src/pkg/packager/deploy_test.go

```diff
@@ -0,0 +1,215 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+package packager
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/defenseunicorns/zarf/src/types"
+)
+
+func TestGenerateValuesOverrides(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name           string
+		chartVariables []types.ZarfChartVariable
+		setVariableMap map[string]*types.ZarfSetVariable
+		deployOpts     types.ZarfDeployOptions
+		componentName  string
+		chartName      string
+		want           map[string]any
+	}{
+		{
+			name:           "Empty inputs",
+			chartVariables: []types.ZarfChartVariable{},
+			setVariableMap: map[string]*types.ZarfSetVariable{},
+			deployOpts:     types.ZarfDeployOptions{},
+			componentName:  "",
+			chartName:      "",
+			want:           map[string]any{},
+		},
+		{
+			name:           "Single variable",
+			chartVariables: []types.ZarfChartVariable{{Name: "testVar", Path: "testVar"}},
+			setVariableMap: map[string]*types.ZarfSetVariable{"testVar": {Value: "testValue"}},
+			deployOpts:     types.ZarfDeployOptions{},
+			componentName:  "testComponent",
+			chartName:      "testChart",
+			want:           map[string]any{"testVar": "testValue"},
+		},
+		{
+			name:           "Non-matching setVariable",
+			chartVariables: []types.ZarfChartVariable{{Name: "expectedVar", Path: "path.to.expectedVar"}},
+			setVariableMap: map[string]*types.ZarfSetVariable{"unexpectedVar": {Value: "unexpectedValue"}},
+			deployOpts:     types.ZarfDeployOptions{},
+			componentName:  "testComponent",
+			chartName:      "testChart",
+			want:           map[string]any{},
+		},
+		{
+			name: "Nested 3 level setVariableMap",
+			chartVariables: []types.ZarfChartVariable{
+				{Name: "level1.level2.level3Var", Path: "level1.level2.level3Var"},
+			},
+			setVariableMap: map[string]*types.ZarfSetVariable{
+				"level1.level2.level3Var": {Value: "nestedValue"},
+			},
+			deployOpts:    types.ZarfDeployOptions{},
+			componentName: "nestedComponent",
+			chartName:     "nestedChart",
+			want: map[string]any{
+				"level1": map[string]any{
+					"level2": map[string]any{
+						"level3Var": "nestedValue",
+					},
+				},
+			},
+		},
+		{
+			name: "Multiple variables with nested and non-nested paths",
+			chartVariables: []types.ZarfChartVariable{
+				{Name: "nestedVar.level2", Path: "nestedVar.level2"},
+				{Name: "simpleVar", Path: "simpleVar"},
+			},
+			setVariableMap: map[string]*types.ZarfSetVariable{
+				"nestedVar.level2": {Value: "nestedValue"},
```

Comment:
```suggestion
				"NESTED_VAR_LEVEL2": {Value: "nestedValue"},
```
Variables would be more accurate here if they matched the pattern that is validated for.

Reply:
I took care of it! Thanks